core/services/chainlink/config_dump_test.go

```diff
@@ -0,0 +1,65 @@
+// Note: These tests are isolated in a separate package and never run in parallel, since they modify they environment.
+
+package chainlink_test
+
+import (
+	"embed"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/kylelemons/godebug/diff"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/smartcontractkit/chainlink/core/internal/testutils"
+	"github.com/smartcontractkit/chainlink/core/services/chainlink"
+)
+
+//go:embed testdata/dump/*.toml
+//go:embed testdata/dump/*.env
+var dumpTestFiles embed.FS
+
+func TestChainlinkApplication_ConfigDump(t *testing.T) {
+	fes, err := dumpTestFiles.ReadDir("testdata/dump")
+	require.NoError(t, err)
+	for _, fe := range fes {
+		if fe.IsDir() {
+			continue
+		}
+		if filepath.Ext(fe.Name()) != ".toml" {
+			continue
+		}
+		name := strings.TrimSuffix(fe.Name(), ".toml")
+		t.Run(name, func(t *testing.T) {
+			exp, err := dumpTestFiles.ReadFile(filepath.Join("testdata/dump", fe.Name()))
+			require.NoError(t, err)
+
+			env, err := dumpTestFiles.ReadFile(filepath.Join("testdata/dump", name) + ".env")
+			require.NoError(t, err)
+
+			//TODO look for optional db file https://app.shortcut.com/chainlinklabs/story/33621/create-config-dump-functionality
+
+			os.Clearenv()
```

Comment:
Note we have this now, can it help? https://github.com/golang/go/issues/41260

Reply:
Maybe could be used, but not currently problematic in our case. I'll check it out.